.github/workflows/pr.yml

```
@@ -117,3 +117,31 @@ jobs:
       uses: actions/upload-artifact@v3
       with:
         path: ./**/*.apk
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3.12.0
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Mock google-services.json
+        run: |
+          cp .github/mock-google-services.json app/google-services.json
+          cp .github/mock-google-services.json wear/google-services.json
+          cp .github/mock-google-services.json automotive/google-services.json
+
+      - name: Run Tests
+        run: ./gradlew test
```

CI testing works, but is taking some time. I don't think you need to test release builds? The main PR build also only uses debug, and there is basically no debug-specific code.

```suggestion
        run: ./gradlew testDebugUnitTest
```